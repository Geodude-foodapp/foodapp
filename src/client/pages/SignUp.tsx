import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import UserForm from '../components/UserForm';

export default ({ logIn }: { logIn: () => void }) => {
  return (
    <section id='signup'>
      <UserForm
        logIn={logIn}
        formMode='Sign Up'
      />
      <p id='signup-prompt'>
        Already have an account?{' '}
        <Link
          to='/login'
          replace
        >
          Log in
        </Link>{' '}
      </p>
    </section>
  );
};
