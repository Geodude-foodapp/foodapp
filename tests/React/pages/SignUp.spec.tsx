import { render } from '@testing-library/react';
import React from 'react';
import SignUp from '../../../src/client/pages/SignUp';

describe('Sign Up Page', () => {
  beforeAll(() => render(<SignUp />));
  it('matches the snapshot', () => {
    const signupPage = document.querySelector('#signup');
    expect(signupPage).toMatchSnapshot();
  });
});
