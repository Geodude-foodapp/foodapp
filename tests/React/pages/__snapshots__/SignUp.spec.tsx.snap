// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Up Page matches the snapshot 1`] = `
<section
  id="signup"
>
  <section
    id="user-form"
  >
    <h2>
      Sign Up Form
    </h2>
    <form>
      <input
        placeholder="username"
        type="text"
        value=""
      />
      <input
        placeholder="password"
        type="password"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
  </section>
</section>
`;
